docs/how_to/memtable-join.md

```diff
@@ -0,0 +1,112 @@
+# How to join an in-memory DataFrame to a TableExpression
+
+You might have an in-memory DataFrame that you want to join to a TableExpression.
+For example, you might have a file on your local machine that you don't want to upload to
+your backend, but you need to join it to a table in that backend.
+
+You can perform joins on local data to TableExpressions from your backend easily with Ibis MemTables.
+
+In this guide, you will lean how to join a pandas DataFrame to a TableExpression.
+
+Data Setup:
+
+In this example, we will create two DataFrames: one containing events and one containing event names.
+We will save the events to a parquet file and read that as a TableExpression in the DuckDB backend.
+We will then convert the event names DataFrame to a PandasInMemoryTable (MemTable), which is
+a pandas DataFrame as a TableExpression and join the two expressions together as we would
+two TableExpressions in a backend.
+
+```python
+    Python 3.10.6 (main, Aug 10 2022, 11:40:04) [GCC 11.3.0]
+    Type 'copyright', 'credits' or 'license' for more information
+    IPython 8.5.0 -- An enhanced Interactive Python. Type '?' for help.
+
+    In [1]: import ibis
+
+    In [2]: import pandas as pd
+       ...: from datetime import date
+
+    In [3]: # create a pandas DataFrame that we will convert to a PandasInMemoryTabl
+       ...: e (Ibis MemTable)
+       ...: events = pd.DataFrame({
+       ...:     'event_id': range(4)
+       ...:     ,'event_name': [f'e{k}' for k in range(4)]
+       ...: })
+
+    In [4]: # Create a parquet file that we will read in using the DuckDB backend
+       ...: # as a TableExpression
+       ...: measures = pd.DataFrame({
+       ...:     "event_id": [0] * 2 + [1] * 3 + [2] * 5 + [3] * 2
+       ...:     ,"measured_on": map(
+       ...:         date
+       ...:         ,[2021] * 12, [6] * 4 + [5] * 6 + [7] * 2
+       ...:         ,range(1, 13)
+       ...:     )
+       ...:     ,"measurement": None
+       ...: })
+
+    In [5]: measures.at[1, "measurement"] = 5.
+       ...: measures.at[4, "measurement"] = 42.
+       ...: measures.at[5, "measurement"] = 42.
+       ...: measures.at[7, "measurement"] = 11.
+
+    In [6]: # Save measures to parquet:
+       ...: measures.to_parquet('data.parquet')
```

Comment:
Maybe this file should more explicitly named be named `"measures.parquet"`:

Reply:
Good idea; I'll update this